src/main.ts

```diff
@@ -178,21 +178,34 @@ ipcMain.handle('login', async (event, host) => {
 
   const handle = await client.deviceAuthorization()
 
-  // eslint-disable-next-line @typescript-eslint/no-floating-promises
-  shell.openExternal(handle.verification_uri_complete)
-
-  // Wait for the user to login.
-  try {
-    console.log('Polling for token')
-    const tokenSet = await handle.poll()
-    console.log('Received token set')
-    console.log(tokenSet)
-    return tokenSet.access_token
-  } catch (e) {
-    console.log(e)
-  }
+  // Register this handle to be used later.
+  ipcMain.handleOnce('loginWithDeviceFlow', async () => {
+    if (!handle) {
+      return Promise.reject(
+        new Error(
+          'No handle available. Did you call startDeviceFlow before calling this?'
+        )
+      )
+    }
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    shell.openExternal(handle.verification_uri_complete)
```

Comment:
Is this okay to be fired off in the background? If so, can we change it to this?
```suggestion
    shell.openExternal(handle.verification_uri_complete).catch(reportRejection)
```

Reply:
I know you didn't write that line, but this is a good time to fix it since we're mucking in here and testing it.

Reply:
Oh yeah I think that you're right, it is fine to fire off in the background. Great catch, thank you